kcbq-connector/src/main/java/com/wepay/kafka/connect/bigquery/write/batch/GCSBatchTableWriter.java

```diff
@@ -0,0 +1,147 @@
+   * @param tableId {@link TableId} of the BigQuery table to upload to
+   * @param bigQuery {@link BigQuery} Object used to perform upload
+   * @param records Records to upload to BigQuery through GCS
+   */
+  public GCSBatchTableWriter(String bucketName,
+                             String blobName,
+                             Storage storage,
+                             TableId tableId,
+                             BigQuery bigQuery,
+                             List<Map<String, Object>> records) {
+    BlobId blobId = BlobId.of(bucketName, blobName + ".json");
+    blobInfo = BlobInfo.newBuilder(blobId).setContentType("text/json").build();
+    this.storage = storage;
+    this.sourceUri = String.format("gs://%s/%s.json", bucketName, blobName);
+
+    this.bigQuery = bigQuery;
+
+    // Check if the table specified exists
+    // This error shouldn't be thrown. All tables should be created by the connector at startup
+    if (bigQuery.getTable(tableId) == null || !bigQuery.getTable(tableId).exists()) {
+      throw new ConnectException("Table with TableId %s does not exist.");
+    }
+
+    this.loadJobConfiguration =
+        LoadJobConfiguration.builder(tableId, sourceUri)
+                            .setFormatOptions(FormatOptions.json())
+                            .setCreateDisposition(JobInfo.CreateDisposition.CREATE_NEVER)
+                            .build();
+
+    this.records = records;
+  }
+
+  @Override
+  public void run() {
+    //todo implement
+  }
+
+  /**
+   * Triggers a load job to transfer JSON records from a GCS blob to a BigQuery table.
+   */
+  private void triggerBigQueryLoadJob() {
+    Job loadJob = bigQuery.create(JobInfo.of(loadJobConfiguration));
+    try {
+      loadJob.waitFor();
+    } catch (InterruptedException | TimeoutException exception) {
+      String exceptionMessage = String.format("%s.Source URI = \"%s\" Table = \"%s\"",
+          "Transfer from GCS blob to BigQuery unsuccessful.",
+          loadJobConfiguration.getSourceUris(),
+          loadJobConfiguration.getDestinationTable());
+      throw new ConnectException(exceptionMessage, exception);
+    }
+  }
+
+  /**
+   * Creates a JSON string containing all records and uploads it as a blob to GCS
+   * @return The blob uploaded to GCS
+   */
+  private Blob uploadRecordsToGcs() {
+    try {
+      return uploadBlobToGcs(new ByteArrayInputStream(toJson(records).getBytes("UTF-8")));
+    } catch (UnsupportedEncodingException uee) {
+      throw new ConnectException("Failed to upload blob to GCS", uee);
+    }
+  }
+
+  private Blob uploadBlobToGcs(InputStream blobContent) {
+    // todo look into creating from a string because this is depreciated - input stream cannot retry
+    // todo consider if it would be worth it to switch to a resumable method of uploading
+    return storage.create(blobInfo, blobContent);
+  }
+
+  private String toJson(Map<String, Object> record) {
+    return gson.toJson(record);
```

Comment:
since this function is just one line and only used in one place, is it really needed?